### PR TITLE
upgrade expo to 37 & re-eject to a Bare workflow

### PR DESCRIPTION
**Pull request by dereckquock**
## upgrade to 37
https://blog.expo.io/expo-sdk-37-is-now-available-dd5770f066a6
https://docs.expo.io/versions/v37.0.0/bare/updating-your-app

## use Bare workflow
https://blog.expo.io/time-to-start-using-expos-bare-workflow-expokit-now-deprecated-d6052890c18b

## use `expo-splash-screen`
https://github.com/expo/expo/tree/master/packages/expo-splash-screen

## use `expo-updates` and import `expo-asset`
https://github.com/expo/expo/tree/master/packages/expo-updates

## add helper button to enable Storybook! 📖 

> notice that blue button with a book icon in it 📚

### CODE REVIEW

File: ios/EchoChurch/AppDelegate.m [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  [super application:application didFinishLaunchingWithOptions:launchOptions];
+
+  return YES;
+}
+
+- (RCTBridge *)initializeReactNativeApp
+{
+  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:self.launchOptions];
+  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge moduleName:@"main" initialProperties:nil];
+  rootView.backgroundColor = [[UIColor alloc] initWithRed:0.0f green:0.0f blue:0.0f alpha:1];
```
Comment (dereckquock) [inline ios/EchoChurch/AppDelegate.m:50]: lol i figured out i had to change this to be `0.0`s

File: ios/EchoChurch/Base.lproj/LaunchScreen.xib [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="15706"/>
+        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
+    </dependencies>
+    <objects>
+        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner"/>
+        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
+        <view contentMode="scaleToFill" id="iN0-l3-epB">
+            <rect key="frame" x="0.0" y="0.0" width="480" height="480"/>
+            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+            <color key="backgroundColor" white="0.0" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
```
Comment (dereckquock) [inline ios/EchoChurch/Base.lproj/LaunchScreen.xib:15]: had to change `white="0.0"`

File: ios/EchoChurch/Supporting/Expo.plist [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+        <key>EXUpdatesSDKVersion</key>
+        <string>YOUR-APP-SDK-VERSION-HERE</string>
```
Comment (dereckquock) [inline ios/EchoChurch/Supporting/Expo.plist]: idk if we need to change this stuff 🤔

File: package.json [hunk 4]
```diff
@@ -1,11 +1,9 @@
 {
-  "main": "node_modules/expo/AppEntry.js",
   "scripts": {
-    "start": "expo start",
-    "android": "expo start --android",
-    "ios": "expo start --ios",
+    "start": "react-native start",
```
Comment (dereckquock) [inline package.json:3]: 😮  it's not expo no more. the Bare workflow is basically using the react-native cli

File: App.js [hunk 5]
```diff
@@ -107,10 +109,36 @@ function App(props) {
         translucent
       />
       <AppNavigator />
+
+      {/* Button to show the Storybook 📖 */}
+      {__DEV__ && (
```
Comment (dereckquock) [inline App.js:114]: this code should get tree shaken out of the main build cause `__DEV__`

File: App.js [hunk 6]
```diff
@@ -74,26 +90,12 @@ function App(props) {
...
 
-  if (!isLoadingComplete && !props.skipLoadingScreen) {
-    return (
-      <AppLoading
-        startAsync={loadResourcesAsync}
-        onError={handleLoadingError}
-        onFinish={handleFinishLoading}
-      />
-    );
+  if (__DEV__ && showStorybook) {
```
Comment (dereckquock) [inline App.js:97]: this code should get tree shaken out of the main build cause `__DEV__`

**Review by Phillip3 — APPROVED** [review]
I APPROVE LOUDLY AND FORCEFULLY

**Pull request merged**